### PR TITLE
Define matching for simple integers, leave others up to implementations

### PR DESCRIPTION
**Pull request by eemeli**
More explicitly defines numeric matching for when it's well-defined, and when it's not well-defined.

Leaves values beyond non-negative integers as implementation-defined, and replaces the JSON reference with the `digit-size-option` ABNF rule, as it's a better fit.

### CODE REVIEW

File: spec/registry.md
Comment (aphillips) [inline spec/registry.md:541]: I think this narrows the range we accept (to omit negative numbers), which isn't strictly necessary. We can repeat the relevant parts of the ABNF as needed.

This also forgets to mention when the `resolvedSelector` matches the `key`, which is kind of important. 

Finally, I would guess that `maximumXXXDigits` options would also be ignored?

**Review by aphillips — COMMENTED** [review]
I agree with removing the JSON reference, but we don't need to get carried away perhaps?

**Comment by aphillips** [comment]
(chair hat ON)

We approved fixing this text in the 2024-11-11 call, so this is on the fast track. Group members should weigh in if they have comments. Note that I require two approvals to merge, per process.

File: spec/registry.md
Comment (eemeli) [inline spec/registry.md:539]:
```suggestion
- `fractionDigits`
- `minimumFractionDigits`
- `minimumIntegerDigits`
- `minimumSignificantDigits`
- `maximumSignificantDigits`
```

File: spec/registry.md
Comment (eemeli) [reply spec/registry.md]: I'm ambivalent about including negative numbers. One ambiguity that opens up is whether `-0` and `0` are distinct.

The `key` comparison is handled in Number Selection.

You're right about missing `maximumSignificantDigits`; added that along with `fractionDigits` for `:currency`. There is no `maximumIntegerDigits`, and `maximumFractionDigits` isn't relevant when the value is an integer.

**Comment by aphillips** [comment]
@macchiati Can I get a review of this? I need two to merge.

**Review by macchiati — CHANGES_REQUESTED** [review]
One minor change needed.

File: spec/registry.md
Comment (macchiati) [reply spec/registry.md:552]: Otherwise ok.

Comment (aphillips) [reply spec/registry.md:552]: Leading zeros (with minimumIntegerDigits) or for fractions, e.g. `100.10` (which doesn't apply to "integer values" of course)

Comment (macchiati) [reply spec/registry.md:552]: It needs to say that, eg

> for integer values without leading zeros.

"non-zero-filled" is an obscure term.

Comment (aphillips) [reply spec/registry.md:552]:
```suggestion
> [!IMPORTANT]
> The exact behavior of exact literal match is only well defined
> for integer values without leading zeros.
```

Comment (aphillips) [reply spec/registry.md:552]:
```suggestion
> for integer values without leading zeros.
```

**Comment by aphillips** [comment]
@macchiati Fixed.

**Review by macchiati — APPROVED** [review]

**Pull request merged**
